src/gradualizer_app.erl

```diff
@@ -15,7 +15,8 @@
 %%%===================================================================
 
 start(_StartType, _StartArgs) ->
-    gradualizer_sup:start_link().
+    Opts = application:get_env(gradualizer, cli_options, []),
```

Comment:
Gradualizer can also run from the Erlang shell and as a rebar3 plugin. These options are not only CLI, are they?

Reply:
Hmm, the real takeaway of this change is that `prelude` default value is now baked into `gradualizer_db` itself, and unless overridden, prelude is imported directly after `init`. Otherwise, we were running into inconsistencies if the DB died and was restarted. Specifically, calling `import_prelude` from the `typechecker` wasn't right, because DB could be restarted before `typechecker` finished checking, and nothing would make the DB import prelude again on restart.

Good point, though, these options are "teleported" here from `gradualizer_cli:main()`, so in that case they are CLI options, but I haven't checked how the Rebar3 plugin runs the rest of the code or how it sets the options. As far as the defaults go, it should be working properly.

Reply:
I've renamed the app env var.

I've also taken a look at `src/rebar_prv_gradualizer.erl`. It seems Gradualizer as an app is not even started from there, which means the plugin relied on `application:ensure_all_started(gradualizer)` being called the first time `typechecker:type_check_forms` was called. This is not completely consistent, though, since the plugin tries to respect that each checked app might have its own set of gradualizer options. However, given the DB is started on the first call to `typechecker`, yet not stopped or reset after the first app is checked, the opts of the second and further checked apps aren't really respected (e.g. prelude won't be unloaded from the already running DB if opts of the second checked app contain `{prelude, false}`). In general, there are some details which need considering and aren't considered currently.

All in all, ultimately we might not want to store just `options`, but actually unpack each option from the list and `set_env` it directly. With that approach the Rebar3 plugin could decide which ought to be set "globally" and which can be overridden per checked app. The plugin could also decide if the Gradualizer app as a whole has to be restarted on each checked app.

Reply:
OK. Do you want to take care of it later or now? I can merge now you're happy with it.

Reply:
Let's leave it for later. The intention of this PR was to unblock CI on other PRs.